### DISCUSSION TITLE
Finding it difficult to use the cmake module - need a bit more help documents to get started

### DISCUSSION THREAD

**Discussion by nareshdevasani** [DEMCON/cmake-sbom#18] [General]
I am trying to understand cmake-sbom for some time but it is difficult to understand what it does(generates sbom but not sure how well it does). I am unable to run this in my project to test the sbom generation.

I have been trying to understand example, but it fails when I try it. And it is mentioned under `Testing` that `This is not required when integrating this repository in your build.` I don't understand, how can I integrate it in my build.

Any help is appreciated.

**Comment by jhrutgers** [comment]
I have added a section to the README with basic instructions how to use it in your project. Does that help?

**Reply by nareshdevasani** [reply]
@jhrutgers can I expect this tool to identify all third party libraries used in the given project and export them in an SBOM file without any change to the given project?

**Reply by jhrutgers** [reply]
No, that's not possible. This project does not do autodetect of libraries, you have to specify in your CMakeLists what should be added to the SBOM.
